cajita/src/Cajita_FastFourierTransform.hpp

```diff
@@ -452,15 +482,38 @@ class HeffteFastFourierTransform
 
         heffte::plan_options heffte_params =
             heffte::default_options<heffte_backend_type>();
-        heffte_params.use_alltoall = params.getAllToAll();
+        // TODO: use all three heffte options for algorithm
+        bool alltoall = params.getAllToAll();
+        if ( alltoall )
+            heffte_params.algorithm = heffte::reshape_algorithm::alltoallv;
+        else
+            heffte_params.algorithm = heffte::reshape_algorithm::p2p;
         heffte_params.use_pencils = params.getPencils();
         heffte_params.use_reorder = params.getReorder();
 
+        // heffte recommends a sycl queue to avoid errors with SYCL backend
+        bool queue = params.getQueue();
+        // TODO: just one call here, no if/else
+
+#ifdef Heffte_ENABLE_ONEAPI
+#ifdef KOKKOS_ENABLE_SYCL
+        sycl::queue q;
+        // Set FFT options from given parameters
+        // heFFTe correctly handles 2D or 3D FFTs within "fft3d"
+        _fft = std::make_shared<heffte::fft3d<heffte_backend_type>>(
```

Comment:
I think the resting environment is trying to compile with one with `heffte_backend_type = mkl`

The issue is that `onemkl` takes a `sycl::queue`, but `mkl` takes either nothing or `nullptr`. This may be the issue.

Reply:
I need to take this PR over and get it up to date with master, then I'll try this suggestion. Thanks!